components/core/src/clp/streaming_archive/single_file_archive/Defs.hpp

```diff
@@ -0,0 +1,84 @@
+#ifndef CLP_STREAMING_ARCHIVE_SINGLE_FILE_ARCHIVE_DEFS_HPP
+#define CLP_STREAMING_ARCHIVE_SINGLE_FILE_ARCHIVE_DEFS_HPP
+
+#include <cstdint>
+#include <string>
+
+#include "../../Defs.h"
+#include "../Constants.hpp"
+#include "msgpack.hpp"
+
+namespace clp::streaming_archive::single_file_archive {
+
+using single_file_archive_format_version_t = uint32_t;
+
+// Single file archive version.
+constexpr uint8_t cArchiveMajorVersion{0};
+constexpr uint8_t cArchiveMinorVersion{1};
+constexpr uint16_t cArchivePatchVersion{1};
+constexpr single_file_archive_format_version_t cArchiveVersion{
+        cArchiveMajorVersion << 24 | cArchiveMinorVersion << 16 | cArchivePatchVersion
+};
+
+static constexpr size_t cNumMagicNumberChars{4};
+static constexpr std::array<uint8_t, cNumMagicNumberChars>
+        cUnstructuredSfaMagicNumber{'Y', 'C', 'L', 'P'};
+static constexpr std::string_view cUnstructuredSfaExtension{".clp"};
+static constexpr size_t cFileSizeWarningThreshold{100L * 1024 * 1024};
+
+static constexpr size_t cNumStaticFiles{5};
+constexpr std::array<char const*, cNumStaticFiles> cStaticArchiveFileNames{
+        cMetadataDBFileName,
+        cLogTypeDictFilename,
+        cLogTypeSegmentIndexFilename,
+        cVarDictFilename,
+        cVarSegmentIndexFilename
+};
+
+static constexpr size_t cNumUnused{6};
+
+struct __attribute__((packed)) SingleFileArchiveHeader {
+    std::array<uint8_t, cNumMagicNumberChars> magic;
+    single_file_archive_format_version_t version;
+    uint64_t metadata_size;
+    std::array<uint64_t, cNumUnused> unused;
+};
+
+struct FileInfo {
+    std::string n;
+    uint64_t o;
+    MSGPACK_DEFINE_MAP(n, o);
+};
+
+struct MultiFileArchiveMetadata {
```

Comment:
@LinZhihao-723 @kirkrodrigues 
My understanding is that we should use class instead of struct, even if it's for data holding purpose. Can you guys confirm?

Reply:
https://google.github.io/styleguide/cppguide.html#Structs_vs._Classes

> Use a struct only for passive objects that carry data; everything else is a class.

> The struct must not have invariants that imply relationships between different fields, since direct user access to those fields may break those invariants.

Do any of the fields in this struct have relationships between each other?

Reply:
Sure, looking at CLP-s, I believe it is consistent